### PR TITLE
Server: Add query parameter validation tests

### PR DESCRIPTION
**Pull request by jaymell**
## Motivation
Add validation tests for query parameters

## Solution
Unit tests asserting valid/invalid usages of params.

### CODE REVIEW

File: server/svix-server/src/v1/endpoints/attempt.rs [hunk 1]
```diff
@@ -627,3 +627,61 @@ pub fn router() -> Router {
...
+    use validator::Validate;
+
+    use super::{ListAttemptedMessagesQueryParameters, ListAttemptsByEndpointQueryParameters};
+
+    #[test]
+    fn test_list_attempted_messages_query_params_validation() {
+        let q: ListAttemptedMessagesQueryParameters = serde_json::from_str(
+            &r#"{
+                "channel": ",invalid",
+                "status": 3
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/attempt.rs:642]: You can just omit it.

File: server/svix-server/src/v1/endpoints/attempt.rs [hunk 2]
```diff
@@ -627,3 +627,61 @@ pub fn router() -> Router {
...
+#[cfg(test)]
+mod tests {
+    use validator::Validate;
+
+    use super::{ListAttemptedMessagesQueryParameters, ListAttemptsByEndpointQueryParameters};
+
+    #[test]
+    fn test_list_attempted_messages_query_params_validation() {
+        let q: ListAttemptedMessagesQueryParameters = serde_json::from_str(
+            &r#"{
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/attempt.rs:640]: Use `json!()`.

File: server/svix-server/src/v1/endpoints/attempt.rs [hunk 3]
```diff
@@ -627,3 +627,61 @@ pub fn router() -> Router {
...
+        )
+        .unwrap();
+        q.validate().unwrap();
+    }
+
+    #[test]
+    fn test_list_attempts_by_endpoint_query_parameters_validation() {
+        let q: ListAttemptsByEndpointQueryParameters = serde_json::from_str(
+            &r#"{
+                "event_types": [ "valid_event_type_name", ",invalid_event_type_name"],
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/attempt.rs:662]: Can you use something more obvious than a leading comma? It almost looks valid. E.g. use $$ or something.

File: server/svix-server/src/v1/endpoints/attempt.rs [hunk 4]
```diff
@@ -627,3 +627,61 @@ pub fn router() -> Router {
...
+                "event_types": [ "valid-event-type-name", "another-valid-event-type-name"],
+                "channel": ",invalid-channel-name"
+            }"#,
+        )
+        .unwrap();
+        assert!(q.validate().is_err());
+
+        let q: ListAttemptsByEndpointQueryParameters = serde_json::from_str(
+            &r#"{
+                "event_types": [ "valid-event-type-name", "another-valid-event-type-name"],
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/attempt.rs:680]: Whitespace is inconsistent.
```suggestion
                "event_types": ["valid-event-type-name", "another-valid-event-type-name"],
```

File: server/svix-server/src/v1/endpoints/attempt.rs (hunk 1 above)
Comment (jaymell) [reply server/svix-server/src/v1/endpoints/attempt.rs:642]: Sorry, omit what?

Comment (tasn) [reply server/svix-server/src/v1/endpoints/attempt.rs:642]: You don't need to include `"status"`, you can just test just `"channel"`. It's optional anyway, so just omitting is fine and makes what you are testing clearer.

**Review by tasn — APPROVED** [review]

**Pull request merged**